src/elements.tsx

```diff
@@ -104,6 +104,8 @@ const isVoidElement = (tagName: string) => {
 };
 
 export function createElement(name: string, attributes: Attributes | undefined, ...contents: string[]) {
+    if (typeof name === 'function') return name(attributes, contents);
```

Comment:
I prefer an `if...else` construct here, with `{}`. That way the intent is more clear for developers.